fix: Add documentation (#1567)
embeddings-benchmark/mteb: docs/api/model.md

# Models

<!-- TODO: Encoder or model? Encoder is consistent with the code, but might be less used WDYT? We also use ModelMeta ... -->

> A model in `mteb` cover two concept, its metadata and its implementation. The metadata contains information about the model such as maximum input
> length, valid frameworks, license and degree of openness. The implementation, is simply a reproducible workflow which allows other to run the
> exact model again using the same prompts, hyperparameters, aggregation strategies and so on.

Reviewer: ```suggestion
A model in `mteb` covers two concepts: metadata and implementation. 
- Metadata contains information about the model such as maximum input
length, valid frameworks, license, and degree of openness. 
- Implementation is a reproducible workflow, which allows others to run the same model again, using the same prompts, hyperparameters, aggregation strategies, etc.
```
